Packet++/src/HttpLayer.cpp

```diff
@@ -890,30 +719,51 @@ void HttpResponseFirstLine::setVersion(HttpVersion newVersion)
 	m_Version = newVersion;
 }
 
-HttpResponseLayer::HttpResponseStatusCode HttpResponseFirstLine::parseStatusCode(const char* data, size_t dataLen)
+HttpResponseStatusCode HttpResponseFirstLine::parseStatusCode(const char* data, size_t dataLen)
 {
 	// minimum data should be 12B long: "HTTP/x.y XXX"
 	if (!data || dataLen < 12)
 	{
-		return HttpResponseLayer::HttpStatusCodeUnknown;
+		return HttpResponseStatusCode::HttpStatusCodeError;
 	}
 
-	const char* statusCodeData = data + 9;
+	std::string statusCodeDataString(data + 9, 3);
 
-	auto codeAsEnum = StatusCodeStringToEnumMap.find(std::string(statusCodeData, 3));
-	if (codeAsEnum == StatusCodeStringToEnumMap.end())
+	for(const auto& pair : statusCodeExplanationStringMap) {
+		if(int(pair.first) == std::stoi(statusCodeDataString)) {
+			return pair.first;
+		}
+	}
```

Comment:
It's not efficient to scan the entire map to find the code, especially because it happens during packet parsing. We should make it more efficient. Maybe convert `statusCodeExplanationStringMap` to `std::unordered_map<int, std::string>`? 🤔

Reply:
let me think about a better way

Reply:
add a `HttpResponseStatusCode` constructor from int, which requires a `unordered_map<int, HttpResponseStatusCode>` in cpp, but I think the performance gain is worth it.